### ISSUE TITLE
TODO ranking 

### ISSUE THREAD

**Issue by Ypot** [CSJ7701/org-ranker#1]
Could it be ranked the org-todo state? 

For example:
#+RANKER-RULE: org-todo==NEXT:3
#+RANKER-RULE: org-todo==WAIT:-1
#+RANKER-RULE: org-todo==DONE:-1000

**Comment by CSJ7701** [comment]
Yes, this should work, since I believe org-mode stores TODO state as a headline property called "TODO".

So your example would be:
#+RANKER-RULE: TODO==NEXT:3
#+RANKER-RULE: TODO==WAIT:-1
#+RANKER-RULE: TODO==DONE:-1000

You can use the function 'org-ranker-list-properties' to show a list of all properties and their values in an org buffer if you're curious about other properties that org-mode uses under the hood.

Edit: Typed the wrong function name.

**Comment by Ypot** [comment]
Hi!!
I can't find TODO states with org-ranker-list-properties. But it would be great.

Best regards.

**Comment by CSJ7701** [comment]
Make sure you've pulled the most recent version - I updated that function in the last commit.

It now takes an optional argument; make sure that argument is passed. That's what makes the built-in org properties show up. I _think_ the universal argument (C-u) should work here, but I haven't had time to verify that yet.

**Comment by Ypot** [comment]
Behaviour is different between `C-u M-x org-ranker-list-properties`, and `(org-ranker-list-properties t)`, but either way no TODO that I can see. Thanks anyway :-)

> Nombre: org-ranker.el
> Tamaño comprimido: 7 693
> Modificado: 2024-12-22 12:57:53
> CRC: E8FC2CD0

**Comment by CSJ7701** [comment]
I just took a look at this - it works for me...

Here's how I tested it, if you want to try this on your end. 
Make sure org-ranker is up to date, then make note of where the `.el` files are.
Using a minimal init file (you can ignore the `(require 'csv ...)` line if you remove the csv dependency, its not necessary for this test):
```emacs-lisp
(require 'csv "path/to/csv.el/csv")
(require 'org-ranker "path/to/org-ranker.el/org-ranker")
```
Now run `emacs -q --load path/to/the/minimal/file.el`.

You'll need a simple org file with some todos. There's a small example in the 'Tests' directory, [here](https://github.com/CSJ7701/org-ranker/blob/main/tests/TodoTest.org).
You're correct that the universal argument isn't working - I plan on fixing this, but for now, you can just call `(org-ranker-list-properties t)`, and you should see something similar to this:

https://github.com/user-attachments/assets/8567aa8f-70ac-482e-b875-7a69c1f11fef

As far as I can tell, this works. If you have any more details about the issue you're having feel free to let me know, but I think this might be an issue on your end.

**Comment by CSJ7701** [comment]
Closing this issue - as far as I can tell the problem is resolved.
